financial-connections/src/main/java/com/stripe/android/financialconnections/features/partnerauth/PartnerAuthViewModel.kt

```diff
@@ -461,18 +464,43 @@ internal class PartnerAuthViewModel @AssistedInject constructor(
         }
     }
 
-    fun onCancelClick() = viewModelScope.launch {
-        // set loading state while cancelling the active auth session, and navigate back
-        setState { copy(authenticationStatus = Loading(value = Status(Action.CANCELLING))) }
-        runCatching {
-            val authSession = requireNotNull(
-                getOrFetchSync(refetchCondition = IfMissingActiveAuthSession).manifest.activeAuthSession
-            )
-            cancelAuthorizationSession(authSession.id)
+    fun onCancelClick() = withState { state ->
+        if (state.inModal) {
+            eventTracker.track(PrepaneClickCancel(pane = PANE))
+        } else {
+            eventTracker.track(PrepaneClickChooseAnotherBank(pane = PANE))
+        }
```

Comment:
One small note - `inModal == relink` is true today, but this can be different as mobile gets to parity with web (there's other scenarios where `partner_auth` is the first pane shown, and in those `inModal == false`

These changes are ok since they're not coupled to relink, but something to note for the future!